components/feature/contextmenu/src/main/java/mozilla/components/feature/contextmenu/ContextMenuCandidate.kt

```diff
@@ -159,6 +164,82 @@ data class ContextMenuCandidate(
             }
         )
 
+        /**
+         * Context Menu item: "Add to contact".
+         */
+        fun createAddContactCandidate(
+            context: Context
+        ) = ContextMenuCandidate(
+            id = "mozac.feature.contextmenu.add_to_contact",
+            label = context.getString(R.string.mozac_feature_contextmenu_add_to_contact),
+            showFor = { _, hitResult -> hitResult.isMailto() },
+            action = { _, hitResult ->
+                val link = hitResult.getLink()
+                val intent = Intent(ContactsContract.Intents.Insert.ACTION).apply {
+                    type = ContactsContract.RawContacts.CONTENT_TYPE
+                    putExtra(ContactsContract.Intents.Insert.EMAIL, link.stripMailToProtocol())
+                    putExtra(
+                        ContactsContract.Intents.Insert.EMAIL_TYPE,
+                        ContactsContract.CommonDataKinds.Email.TYPE_WORK
+                    )
+                    addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+                }
+
+                context.startActivity(intent)
+            }
+        )
+
+        /**
+         * Context Menu item: "Share email address".
+         */
+        fun createShareEmailAddressCandidate(
+            context: Context
+        ) = ContextMenuCandidate(
+            id = "mozac.feature.contextmenu.share_email",
+            label = context.getString(R.string.mozac_feature_contextmenu_share_email_address),
+            showFor = { _, hitResult -> hitResult.isMailto() },
+            action = { _, hitResult ->
+                val intent = Intent(Intent.ACTION_SEND).apply {
+                    type = "text/plain"
+                    flags = Intent.FLAG_ACTIVITY_NEW_TASK
+                    putExtra(Intent.EXTRA_TEXT, hitResult.getLink().stripMailToProtocol())
+                }
+
+                val shareIntent = Intent.createChooser(
+                    intent,
+                    context.getString(R.string.mozac_feature_contextmenu_share_email_address)
+                ).addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+
+                context.startActivity(shareIntent)
```

Comment:
Maybe we can reuse [Context.email](https://github.com/mozilla-mobile/android-components/blob/62efb06f13083d30071849cdc4eb05a4419a7ed5/components/support/ktx/src/main/java/mozilla/components/support/ktx/android/content/Context.kt#L118)

Reply:
I'll look into this one.  I think there was some issue using it but I didn't revisit after completing my prototype.  Thanks

Reply:
Confirmed I can't use it because the action here is to share the email.  But `Context.email` is to compose a new email.
I can also move this to `Context` extension so it can be reused.